### PR TITLE
add easyconfigs for cuDNN 8.1.x and update 8.0.x easyconfigs with a download location

### PR DESCRIPTION
**Pull request by robert-mijakovic**
(created using `eb --new-pr`)

Edit: Requires ~https://github.com/easybuilders/easybuild-easyblocks/pull/2356~ for armsbsa (AARCH64)

### CODE REVIEW

File: easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.0.2.eb [hunk 1]
```diff
@@ -3,28 +3,30 @@
...
+accept_eula = True
+source_urls = ['https://developer.download.nvidia.com/compute/redist/cudnn/v%s/' % '.'.join(version.split('.')[:3])]
+local_tarball_tmpl = '-'.join(['%%(namelower)s', local_cuda_version_majmin, 'linux', '%s', 'v%%(version)s.tgz'])
+sources = [local_tarball_tmpl % '%(cudnnarch)s']
 checksums = [
     {
-        '%%(namelower)s-%s-linux-x64-v%%(version)s.tgz' % local_cuda_version_major_minor:
+        local_tarball_tmpl % 'x64':
             '38a81a28952e314e21577432b0bab68357ef9de7f6c8858f721f78df9ee60c35',
-        '%%(namelower)s-%s-linux-ppc64le-v%%(version)s.tgz' % local_cuda_version_major_minor:
```
Comment (Micket) [inline easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.0.2.eb:27]: Please keep the PPC variant here

File: easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.1.1.eb [hunk 2]
```diff
@@ -3,28 +3,30 @@
...
+accept_eula = True
+source_urls = ['https://developer.download.nvidia.com/compute/redist/cudnn/v%s/' % '.'.join(version.split('.')[:3])]
+local_tarball_tmpl = '-'.join(['%%(namelower)s', local_cuda_version_majmin, 'linux', '%s', 'v%%(version)s.tgz'])
+sources = [local_tarball_tmpl % '%(cudnnarch)s']
 checksums = [
     {
-        '%%(namelower)s-%s-linux-x64-v%%(version)s.tgz' % local_cuda_version_major_minor:
+        local_tarball_tmpl % 'x64':
             '8f4c662343afce5998ce963500fe3bb167e9a508c1a1a949d821a4b80fa9beab',
-        '%%(namelower)s-%s-linux-ppc64le-v%%(version)s.tgz' % local_cuda_version_major_minor:
```
Comment (Micket) [inline easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.1.1.eb:27]: Please keep the PPC variant here

**Review by Micket — CHANGES_REQUESTED** [review]
This seems to remove the ppc64 variants from all the versions instead.

**Comment by robert-mijakovic** [comment]
@Micket Thank you for noticing. This was actually a copy-paste problem. It is fixed.

**Comment by robert-mijakovic** [comment]
@Micket Can you approve this one?

File: easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.0.2.eb [hunk 3]
```diff
@@ -1,30 +1,29 @@
 ##
-# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
-#
-# Author:    Stephane Thiell <sthiell@stanford.edu>
```
Comment (boegel) [inline easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.0.2.eb:4]: @robert-mijakovic It's fine if you add yourself as author, but keep the original one too please.

Comment (robert-mijakovic) [reply easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.0.2.eb:4]: I'm sorry about it.  It wasn't intentional. I will fix it.

File: easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.0.2.eb [hunk 4]
```diff
@@ -1,30 +1,29 @@
...
+a GPU-accelerated library of primitives for deep neural networks."""
 
 toolchain = SYSTEM
 
-# Nvidia developer registration required.
-# Download link: https://developer.nvidia.com/rdp/cudnn-download
-sources = ['%%(namelower)s-%s-linux-%%(cudnnarch)s-v%%(version)s.tgz' % local_cuda_version_major_minor]
+# By downloading, you accept the cuDNN Software License Agreement
+# (https://docs.nvidia.com/deeplearning/sdk/cudnn-sla/index.html)
+accept_eula = True
```
[the remark this answers is not in the document]
Comment (robert-mijakovic) [reply easybuild/easyconfigs/c/cuDNN/cuDNN-8.0.4.30-CUDA-11.0.2.eb]: Fixed!

**Comment by robert-mijakovic** [comment]
@boegel I forgot to mention that it is done.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
mgmt.select-pika.citc.local - Linux centos linux 8.3.2011, x86_64, AMD EPYC 7571, Python 3.6.8
See https://gist.github.com/6c1b689c248d493827b087bd22281602 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
easybuild2.novalocal - Linux centos linux 8.3.2011, POWER, IBM pSeries (emulated by qemu) (power9le), Python 3.6.8
See https://gist.github.com/0d5a72d027915f9d4fdcd09be5604c28 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @robert-mijakovic!

**Pull request merged**